tests/Integration/RestTest.php

```diff
@@ -126,6 +149,62 @@ public function test_get_callback(): void {
 		self::assertEquals( $expected, $meta_object );
 	}
 
+	/**
+	 * Test that the rendered meta function returns the meta HTML string with json ld.
+	 *
+	 * @covers \Parsely\Rest::get_rendered_meta
+	 */
+	public function test_get_rendered_meta_json_ld(): void {
+		// Set the default options prior to each test.
+		TestCase::set_options();
+
+		global $post;
+		$post_id = self::factory()->post->create();
+
+		// phpcs:ignore WordPress.WP.GlobalVariablesOverride.Prohibited
+		$post = get_post( $post_id );
+		$date = gmdate( 'Y-m-d\TH:i:s\Z', get_post_time( 'U', true, $post ) );
+
+		$meta_string = self::$rest->get_rendered_meta();
+		$expected    = '
+<script type="application/ld+json">
+{"@context":"http:\/\/schema.org","@type":"NewsArticle","mainEntityOfPage":{"@type":"WebPage","@id":"http:\/\/example.org\/?p=22"},"headline":"Post title 0000054","url":"http:\/\/example.org\/?p=22","thumbnailUrl":"","image":{"@type":"ImageObject","url":""},"dateCreated":"' . $date . '","datePublished":"' . $date . '","dateModified":"' . $date . '","articleSection":"Uncategorized","author":[],"creator":[],"publisher":{"@type":"Organization","name":"Test Blog","logo":""},"keywords":[]}
+</script>
+
+';
+		self::assertEquals( $expected, $meta_string );
+	}
+
+	/**
+	 * Test that the rendered meta function returns the meta HTML string with json ld.
+	 *
+	 * @covers \Parsely\Rest::get_rendered_meta
+	 */
+	public function test_get_rendered_repeated_metas(): void {
+		// Set the default options prior to each test.
+		TestCase::set_options(
+			array( 'meta_type' => 'repeated_metas' )
+		);
+
+		global $post;
+		$post_id = self::factory()->post->create();
+
+		// phpcs:ignore WordPress.WP.GlobalVariablesOverride.Prohibited
+		$post = get_post( $post_id );
+		$date = gmdate( 'Y-m-d\TH:i:s\Z', get_post_time( 'U', true, $post ) );
+
+		$meta_string = self::$rest->get_rendered_meta();
+		$expected    = '
+<meta name="parsely-title" content="Post title 0000055" />
+<meta name="parsely-link" content="http://example.org/?p=23" />
```

Comment:
This assumes that the tests are run in the same order, which they aren't. So the failure is a flaw in the test logic, not the active code.

The post title can be set when creating the post with the factory, or retrieved from `$post_id`.
The `?p=` value can use `$post_id`.

Reply:
Fixed in [this commit](https://github.com/Parsely/wp-parsely/pull/500/commits/504fc6d5b13cbbfbed5af225c8cfa6c96df68329).